Loop/Managers/LoopDataManager.swift

```diff
@@ -397,13 +397,9 @@ final class LoopDataManager: ObservableObject {
         }
 
         var overrides = temporaryPresetsManager.overrideHistory.getOverrideHistory(startDate: neededSensitivityTimeline.start, endDate: forecastEndTime)
-
-        // Bug (https://tidepool.atlassian.net/browse/LOOP-4759) pre-meal is not recorded in override history
-        // So currently we handle automatic forecast by manually adding it in, and when meal bolusing, we do not do this.
-        // Eventually, when pre-meal is stored in override history, during meal bolusing we should scan for it and adjust the end time
-        if !disablingPreMeal, let preMeal = temporaryPresetsManager.preMealOverride {
-            overrides.append(preMeal)
-            overrides.sort { $0.startDate < $1.startDate }
+        
+        if disablingPreMeal, let preMealOverride = temporaryPresetsManager.preMealOverride, let index = overrides.firstIndex(where: { $0 == preMealOverride }) {
```

Comment:
Maybe last index? In case there are more in the history?  We want to disable a *currently running* one. Maybe check that you find a currently running one; i.e. one not ended at baseTime.